File: app/src/main/java/com/tananaev/logcat/StringUtils.java
package com.tananaev.logcat;

import android.support.annotation.Nullable;

public abstract class StringUtils {

    private static final int INDEX_NOT_FOUND = -1;

    public static int indexOfIgnoreCase(@Nullable String str, @Nullable String searchStr) {
        return indexOfIgnoreCase(str, searchStr, 0);
    }

    public static boolean containsIgnoreCase(@Nullable String str, @Nullable String searchStr) {
        return indexOfIgnoreCase(str, searchStr, 0) >= 0;
    }

    public static int indexOfIgnoreCase(@Nullable String str, @Nullable String searchStr, int startPos) {
        if (str == null || searchStr == null) {
            return INDEX_NOT_FOUND;
        }
        if (startPos < 0) {
            startPos = 0;
        }
        final int endLimit = str.length() - searchStr.length() + 1;
        if (startPos > endLimit) {
            return INDEX_NOT_FOUND;
        }
        if (searchStr.length() == 0) {
            return startPos;
        }
        for (int i = startPos; i < endLimit; i++) {
            if (str.regionMatches(true, i, searchStr, 0, searchStr.length())) {
                return i;
            }
        }
        return INDEX_NOT_FOUND;
    }
}

File: app/src/main/java/com/tananaev/logcat/StringUtils.kt
/*
 * Copyright 2016 - 2022 Anton Tananaev (anton.tananaev@gmail.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.tananaev.logcat

object StringUtils {

    private const val INDEX_NOT_FOUND = -1

    @JvmStatic
    fun containsIgnoreCase(str: String?, searchStr: String?): Boolean {
        return indexOfIgnoreCase(str, searchStr, 0) >= 0
    }

    @JvmStatic
    fun indexOfIgnoreCase(str: String?, searchString: String?, startPosition: Int = 0): Int {
        var startIndex = startPosition
        if (str == null || searchString == null) {
            return INDEX_NOT_FOUND
        }
        if (startIndex < 0) {
            startIndex = 0
        }
        val endLimit = str.length - searchString.length + 1
        if (startIndex > endLimit) {
            return INDEX_NOT_FOUND
        }
        if (searchString.isEmpty()) {
            return startIndex
        }
        for (i in startIndex until endLimit) {
            if (str.regionMatches(i, searchString, 0, searchString.length, ignoreCase = true)) {
                return i
            }
        }
        return INDEX_NOT_FOUND
    }

}
